main.py

```diff
@@ -37,118 +96,49 @@ def apply_distance_calc(x, lat, lon):
     return calculate_distance(lat, lon, x.latitude, x.longitude)
 
 
-def main():
-    start_time = time.perf_counter()
-    lat = start_lat
-    lon = start_lon
-    distances_df = pd.DataFrame()
-    travel_df = pd.DataFrame(columns=["brewery_id", "distance"])
-    current_distance = 0
-    total_distance = 0
-    while True:
+def calculate_ratio(row):
+    return (row.beer_count + 1) / (row.distance + 0.01)
 
-        # Apply the distance calculation formula
-        distances_df = (
-            geocodes_df.query("Visited == False")
-            .apply(apply_distance_calc, args=(lat, lon), axis=1)
-            .rename("distances")
-            .to_frame()
-        )
 
-        # Check if there's at least one location that is closer than MAX_DISTANCE
-        # MAX_DISTANCE is divided by 2, since you'll need to come back to starting location
-        if distances_df.min().values[0] > (MAX_DISTANCE / 2):
-            print(
-                f"Sorry, no breweries within {MAX_DISTANCE}km from this starting location."
-            )
-            print(f"\nProgram took: {time.perf_counter() - start_time} seconds")
-            break
+def get_next_location(distances_df, km_left):
+    distances_df["distance_to_home"] = distances_df.apply(
+        apply_distance_calc, args=(start_lat, start_lon), axis=1
+    )
+    distances_df["ratio"] = distances_df.apply(calculate_ratio, axis=1)
+    best_next_location = pd.Series(distances_df.query("distance < 1000").ratio.idxmax())
```

Comment:
I think `distances_df.query("distance < 1000")` could be extracted to a separate line for readability.
Also, put it before "ratio" line. As we do not need to calculate ratio for locations out of reach.

Reply:
Second, though, this line is not needed, as the problem that it is dealing with should be handled by km_left  argument